File: judge-protocol/src/main/scala/judgeprotocol/objects/request/ReportHackResultRequest.scala
package judgeprotocol.objects.request

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
import judgeprotocol.objects.response.JudgeResult

/** judger 完成 hack 尝试后回报的状态、可选答案、结果快照和诊断消息。 */
final case class ReportHackResultRequest(
  status: String,
  answer: Option[String],
  newResult: Option[JudgeResult],
  validatorMessage: Option[String],
  standardMessage: Option[String],
  targetMessage: Option[String]
)

/** 负责 hack 结果回报请求的协议编解码；status 仍由 backend HackStatus 解析校验。 */
object ReportHackResultRequest:
  given Encoder[ReportHackResultRequest] = deriveEncoder[ReportHackResultRequest]
  given Decoder[ReportHackResultRequest] = deriveDecoder[ReportHackResultRequest]

File: frontend/src/objects/hack/request/ReportHackResultRequest.ts
import type { HackStatus } from '@/objects/hack/HackStatus'
import type { JudgeResult } from '@/objects/submission/JudgeResult'

/** Judger 完成 hack attempt 后回报的结果载荷；对象对齐例外：镜像 judge-protocol ReportHackResultRequest，不对应后端 hack domain object。 */
export type ReportHackResultRequest = {
  status: HackStatus
  answer: string | null
  newResult: JudgeResult | null
  validatorMessage: string | null
  standardMessage: string | null
  targetMessage: string | null
}
